docs/examples/api-reference/operators/summarize.py

```diff
@@ -0,0 +1,119 @@
+import unittest
+from datetime import datetime
+
+import pandas as pd
+from fennel.testing import mock
+
+__owner__ = "hoang@fennel.ai"
+
+
+class TestAssignSnips(unittest.TestCase):
+    @mock
+    def test_basic(self, client):
+        # docsnip basic
+        from fennel.datasets import (
+            dataset,
+            field,
+            pipeline,
+            Dataset,
+        )
+        from fennel.lib import inputs
+        from fennel.sources import source, Webhook
+        from fennel.dtypes import Window
+
+        webhook = Webhook(name="webhook")
+
+        @source(webhook.endpoint("Transaction"), disorder="14d", cdc="append")
+        @dataset
+        class Transaction:
+            uid: int
+            amount: int
+            timestamp: datetime = field(timestamp=True)
+
+        @dataset
+        class Session:
+            # docsnip-highlight start
+            # groupby field becomes the key field
+            uid: int = field(key=True)
+            # new fields are added to the dataset by the window operation
+            window: Window = field(key=True)
+            total_amount: int
+            # docsnip-highlight end
+            timestamp: datetime = field(timestamp=True)
+
+            @pipeline
+            @inputs(Transaction)
+            def window_pipeline(cls, ds: Dataset):
+                # docsnip-highlight start
+                return (
+                    ds.groupby("uid")
+                    .window(type="session", gap="15m", field="window")
```

Comment:
I don't think `field` as kwarg is clear enough - could we change this to make it more human readable in english like a sentence? If nothing else, we should use `into_field` since that's what we use for aggregations (though i don't like even that).

Reply:
btw I don't feel as strongly about this as I do about `summarize` feeling similar to `assign` in how it's written.

Reply:
I think this is reasonable !